src/genpy/rostime.py

```diff
@@ -395,6 +396,11 @@ def __floordiv__(self, val):
         """
         t = type(val)
         if t in (int, long):
+            warnings.warn(
+                'The Duration.__floordiv__(integer) function is ill-defined. '
+                'The floor operation is applied independently on the seconds as well as the nanoseconds. '
+                'For true division use the operator `/` (which requires `from __future__ import division` in Python 2) or `__truediv__` instead.',
```

Comment:
This seems to advise direct use of `__truediv__`, which is not recommended. `operator.truediv` is a better substitute here, because that can correctly deal with `NotImplemented` and `__rdiv__`.